protocol/linkpreview_unfurler_status.go

```diff
@@ -0,0 +1,187 @@
+package protocol
+
+import (
+	"fmt"
+
+	"go.uber.org/zap"
+
+	"github.com/status-im/status-go/api/multiformat"
+	"github.com/status-im/status-go/images"
+	"github.com/status-im/status-go/protocol/common"
+	"github.com/status-im/status-go/protocol/communities"
+)
+
+type StatusUnfurler struct {
+	m      *Messenger
+	logger *zap.Logger
+	url    string
+}
+
+func NewStatusUnfurler(URL string, messenger *Messenger, logger *zap.Logger) *StatusUnfurler {
+	return &StatusUnfurler{
+		m:      messenger,
+		logger: logger,
+		url:    URL,
+	}
+}
+
+func buildThumbnail(image *images.IdentityImage, thumbnail *common.LinkPreviewThumbnail) error {
```

Comment:
mb `updateThumbnail` if it's not in the arguments

Reply:
I second `updateThumbnail`, because `build` implies the function is usually immutable.